mne/encoding/model.py

```diff
@@ -0,0 +1,323 @@
+"""Base and high-level classes for fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from .feature import EventsBinarizer, DataDelayer
+from ..io.pick import pick_types, pick_info
+from ..externals.six import string_types
+from ..evoked import EvokedArray
+
+
+class EventRelatedRegressor(object):
+    """Fit an rER[R/F] model to Raw objects.
+
+    This uses event onsets to create a continuous, binary representation of
+    event times. It creates a time-lagged version of these events, in order to
+    fit a set coefficients per event type that represent the effect of that
+    event on the signal of interest.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The data to be modeled with event onsets. Data must be preloaded.
+    events : array, shape (n_events, 3)
+        An MNE events array specifying event onset indices (first column),
+        and event types (last column)
+    est : None | instance of sklearn estimator | string
+        A sklearn-style estimator. The input matrix will be a binary matrix
+        of events, while the output will be the neural signal of each channel.
+        If a string, then an instance of a sklearn `Ridge` model will be
+        created with alpha == 0, and this parameter passed to the `solver`
+        parameter.
+    event_id : dictionary
+        A dictionary of (event_id: event_num) pairs
+    tmin : float
+        The minimum time for the rER[P/F]
```

Comment:
this would be pretty easy to do but it might get confusing keeping track of which new features belong to which events. We could say that either `delays` or `time_window` could also be a dictionary of `event_id: value` pairs or something.